### PR TITLE
HDDS-2137 : OzoneUtils to verify resourceName using HddsClientUtils

### PR DESCRIPTION
**Pull request by virajjasani**

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 1184 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 72 | Maven dependency ordering for branch |
| -1 | mvninstall | 31 | hadoop-ozone in trunk failed. |
| -1 | compile | 22 | hadoop-ozone in trunk failed. |
| +1 | checkstyle | 62 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 927 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 168 | trunk passed |
| 0 | spotbugs | 204 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 26 | hadoop-ozone in trunk failed. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 27 | Maven dependency ordering for patch |
| -1 | mvninstall | 34 | hadoop-ozone in the patch failed. |
| -1 | compile | 25 | hadoop-ozone in the patch failed. |
| -1 | javac | 25 | hadoop-ozone in the patch failed. |
| -0 | checkstyle | 28 | hadoop-hdds: The patch generated 7 new + 38 unchanged - 5 fixed = 45 total (was 43) |
| -0 | checkstyle | 28 | hadoop-ozone: The patch generated 3 new + 26 unchanged - 9 fixed = 29 total (was 35) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 723 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 166 | the patch passed |
| -1 | findbugs | 25 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 160 | hadoop-hdds in the patch failed. |
| -1 | unit | 27 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 31 | The patch does not generate ASF License warnings. |
| | | 4616 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.container.keyvalue.TestKeyValueContainer |
|   | hadoop.ozone.container.ozoneimpl.TestOzoneContainer |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1455 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux eee188862a63 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 56f042c |
| Default Java | 1.8.0_222 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/1/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/1/artifact/out/branch-compile-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/1/artifact/out/branch-findbugs-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/1/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/1/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/1/artifact/out/patch-compile-hadoop-ozone.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/1/artifact/out/diff-checkstyle-hadoop-hdds.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/1/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/1/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/1/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/1/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/1/testReport/ |
| Max. process+thread count | 340 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/client hadoop-ozone/common U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by virajjasani** [comment]
@elek @adoroszlai Could you please let me know if there is a way to build hadoop-ozone with hadoop-hdds client dependency?

**Comment by elek** [comment]
@virajjasani I am not sure what do you need exactly. hadoop-ozone always use hadoop-hdds (common, client as a dependency).

The alternative CI builds is not triggered because we have a [whitelist](https://github.com/elek/argo-ozone/commit/6bd2ebe1ae78c55ae1c5b2e6c30b407443fb1b13) to allow PR builds

**Comment by virajjasani** [comment]
> @virajjasani I am not sure what do you need exactly. hadoop-ozone always use hadoop-hdds (common, client as a dependency).
> 
> The alternative CI builds is not triggered because we have a [whitelist](https://github.com/elek/argo-ozone/commit/6bd2ebe1ae78c55ae1c5b2e6c30b407443fb1b13) to allow PR builds

Thanks @elek

File: hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/client/HddsClientUtils.java [hunk 1]
```diff
@@ -126,75 +126,63 @@ public static long formatDateTime(String date) throws ParseException {
...
+          "Bucket or Volume length is illegal, valid length is 3-63 characters");
     }
 
-    if ((resName.charAt(0) == '.') || (resName.charAt(0) == '-')) {
+    if (resName.charAt(0) == '.' || resName.charAt(0) == '-') {
       throw new IllegalArgumentException(
           "Bucket or Volume name cannot start with a period or dash");
     }
 
     if ((resName.charAt(resName.length() - 1) == '.') ||
```
Comment (bharatviswa504) [inline hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/client/HddsClientUtils.java]: I see the braces are removed everywhere, to be consistent can we remove from here too?

Comment (bharatviswa504) [reply hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/client/HddsClientUtils.java]: `if (resName.charAt(resName.length() - 1) == '-' || resName.charAt(resName.length() - 1) == '-') {`

Comment (virajjasani) [reply hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/client/HddsClientUtils.java]: Done. Thanks for the review @bharatviswa504

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 83 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 68 | Maven dependency ordering for branch |
| -1 | mvninstall | 31 | hadoop-ozone in trunk failed. |
| -1 | compile | 20 | hadoop-ozone in trunk failed. |
| +1 | checkstyle | 46 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 882 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 151 | trunk passed |
| 0 | spotbugs | 174 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| -1 | findbugs | 24 | hadoop-ozone in trunk failed. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 26 | Maven dependency ordering for patch |
| -1 | mvninstall | 32 | hadoop-ozone in the patch failed. |
| -1 | compile | 22 | hadoop-ozone in the patch failed. |
| -1 | javac | 22 | hadoop-ozone in the patch failed. |
| -0 | checkstyle | 26 | hadoop-hdds: The patch generated 8 new + 37 unchanged - 6 fixed = 45 total (was 43) |
| -0 | checkstyle | 25 | hadoop-ozone: The patch generated 3 new + 26 unchanged - 9 fixed = 29 total (was 35) |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 705 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 153 | the patch passed |
| -1 | findbugs | 23 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| +1 | unit | 282 | hadoop-hdds in the patch passed. |
| -1 | unit | 25 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 28 | The patch does not generate ASF License warnings. |
| | | 3400 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.2 Server=19.03.2 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1455 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux f63c9bb8183d 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 01d7924 |
| Default Java | 1.8.0_212 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/2/artifact/out/branch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/2/artifact/out/branch-compile-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/2/artifact/out/branch-findbugs-hadoop-ozone.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/2/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/2/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/2/artifact/out/patch-compile-hadoop-ozone.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/2/artifact/out/diff-checkstyle-hadoop-hdds.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/2/artifact/out/diff-checkstyle-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/2/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/2/testReport/ |
| Max. process+thread count | 380 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/client hadoop-ozone/common U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1455/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**